sound/soc/sof/intel/hda.c

```diff
@@ -1193,12 +1193,14 @@ static bool link_slaves_found(struct snd_sof_dev *sdev,
 		mfg_id = SDW_MFG_ID(adr);
 		part_id = SDW_PART_ID(adr);
 		link_id = SDW_DISCO_LINK_ID(adr);
+		version = SDW_VERSION(adr);
 
 		for (j = 0; j < num_slaves; j++) {
 			/* find out how many identical parts were reported on that link */
 			if (ids[j].link_id == link_id &&
 			    ids[j].id.part_id == part_id &&
-			    ids[j].id.mfg_id == mfg_id)
+			    ids[j].id.mfg_id == mfg_id &&
```

Comment:
@bardliao code looks OK but a few typos in the commit message, "metch", "differnt". Also, I was wondeirng this check here is repeated in at least 2 other places, would it make sense to add a macro or a helper?

Reply:
> @bardliao code looks OK but a few typos in the commit message, "metch", "differnt". Also, I was wondeirng this check here is repeated in at least 2 other places, would it make sense to add a macro or a helper?

The typo are fixed. The checks are similar but not the same. The first and second check is to compare ids and adr, the last one is to compare two adrs. So I will leave it as it is.